src/traces/funnelarea/plot.js

```diff
@@ -0,0 +1,306 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+
+var Drawing = require('../../components/drawing');
+var Lib = require('../../lib');
+var svgTextUtils = require('../../lib/svg_text_utils');
+
+var barPlot = require('../bar/plot');
+var getTransformToMoveInsideBar = barPlot.getTransformToMoveInsideBar;
+
+var pieHelpers = require('../pie/helpers');
+var piePlot = require('../pie/plot');
+
+var attachFxHandlers = piePlot.attachFxHandlers;
+var determineInsideTextFont = piePlot.determineInsideTextFont;
+
+var scalePies = piePlot.scalePies;
+var prerenderTitles = piePlot.prerenderTitles;
+var positionTitleOutside = piePlot.positionTitleOutside;
+
+module.exports = function plot(gd, cdModule) {
+    var fullLayout = gd._fullLayout;
+
+    prerenderTitles(cdModule, gd);
+    scalePies(cdModule, fullLayout._size);
+
+    Lib.makeTraceGroups(fullLayout._funnelarealayer, cdModule, 'trace').each(function(cd) {
+        var plotGroup = d3.select(this);
+        var cd0 = cd[0];
+        var trace = cd0.trace;
+
+        setCoords(cd);
+
+        // TODO: miter might look better but can sometimes cause problems
+        // maybe miter with a small-ish stroke-miterlimit?
+        plotGroup.attr('stroke-linejoin', 'round');
+
+        plotGroup.each(function() {
+            var slices = d3.select(this).selectAll('g.slice').data(cd);
+
+            slices.enter().append('g')
+                .classed('slice', true);
+            slices.exit().remove();
+
+            var quadrants = [
```

Comment:
`quadrant` isn't used anywhere for `funnelarea`. You can  :hocho: it

Reply:
Cut in https://github.com/plotly/plotly.js/pull/3876/commits/7e6b9aee160110a8d17f3be746f6ccf44531501d.